### PR TITLE
FEM-2235 Send AdClickedEvent support + move AdRequestedEvent once AdManager is loaded

### PR DESCRIPTION
**Pull request by giladna**
move AdRequested Event so Youbora Ad Manager  will receive it

add null protection before calling getPlayer

### CODE REVIEW

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java [hunk 1]
```diff
@@ -1079,7 +1080,21 @@ public void run() {
...
+                    messageBus.post(new AdEvent.AdClickedEvent(clickThruUrl));
+                    return;
+                } catch (NoSuchMethodException e) {
+                    e.printStackTrace();
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                } catch (InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                messageBus.post(new AdEvent.AdClickedEvent("Unavailable"));
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java]: Better use `null` instead of `"Unavailable"`.

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java (hunk 1 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java]: ok

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java [hunk 2]
```diff
@@ -1079,7 +1080,21 @@ public void run() {
                 break;
             case CLICKED:
                 isAdIsPaused = true;
-                messageBus.post(new AdEvent(AdEvent.Type.CLICKED));
+                String clickThruUrl = null;
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java]: Consider extracting this logic to a new method `handleAdClicked()`. IMHO it's too long for a `case`.

Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java]: done

**Review by noamtamim — APPROVED** [review]

**Pull request merged**